collections/_article/introducing-the-improved-ufbx-importer-in-godot-4-3.md

```diff
@@ -0,0 +1,55 @@
+---
+title: "Introducing the improved ufbx importer in Godot 4.3"
+excerpt: "Godot 4.3 now includes native .fbx support"
+categories: ["progress-report"]
+author: Lyuma
+image: /storage/blog/covers/PlaceHolder.webp
+date: 2099-09-09 00:00:00
+---
+
+Thanks to [a game-changing contribution by Samuli Raivio (bqqbarbhg) and Ernest Lee (fire)](https://github.com/godotengine/godot/pull/81746), Godot 4.3 now includes native .fbx support. This was made possible using Samuli’s free and open source [ufbx single-file fbx library(https://github.com/ufbx/ufbx). The ufbx importer will be used by default for all newly imported fbx files in Godot 4.3
+
+# What’s new in ufbx?
+
+First and foremost, it will no longer be necessary to download the external FBX2glTF converter in order to use .fbx files in Godot 4.3 projects. This also enables .fbx support on all platforms Godot supports, as well as script access using the new FBXDocument API.
+
+The new importer also contains many technical improvements and changes:
+* Imported scenes no longer have an extraneous "RootNode" top-level node in the scene.
+* Unit conversion (such as centimeters to meters)  is now applied to the mesh itself rather than assigned to the node scale of top-level nodes.
+* Improved support for PBR materials and lighting accuracy, within what the .fbx format supports. (.fbx tends to have inconsistent material support between applications)
+* Support for geometric pivots. Pivot offsets are applied automatically where possible.
+* The option to create geometric helper nodes helps support pivots and inheritance modes not otherwise possible with traditional translation, rotation and scaling node transforms.
+* Automatic trimming of animations based on the start and end timestamp encoded in the .fbx file header.
+* Preserves the original material order from the DCC, rather than sorted by face index.
+* Calculates an accurate bone rest using information such as the bind-pose. This can assist with skeleton retargeting.
+
+# Improved import of animated characters
+
+Those who have worked with .fbx in the past may have encountered a plethora of issues when importing skeletons and animations, to the extent that it became common practice to import such animation files in Blender and re-export as glTF 2.0. WIth Godot 4.3, we hope that this will be a thing of the past. We have made the following improvements to the skeleton importer:
```

Comment:
```suggestion
Those who have worked with .fbx in the past may have encountered a plethora of issues when importing skeletons and animations, to the extent that it became common practice to import such animation files in Blender and re-export as glTF 2.0. With Godot 4.3, we hope that this will be a thing of the past. We have made the following improvements to the skeleton importer:
```